vm/actor/src/builtin/reward/logic.rs

```diff
@@ -0,0 +1,94 @@
+
+lazy_static! {
+    /// Floor(e^(ln[1 + 200%] / epochsInYear) * 2^128
+    /// Q.128 formatted number such that f(epoch) = baseExponent^epoch grows 200% in one
+    /// year of epochs
+    /// Calculation here: https://www.wolframalpha.com/input/?i=IntegerPart%5BExp%5BLog%5B1%2B200%25%5D%2F%28%28365+days%29%2F%2830+seconds%29%29%5D*2%5E128%5D
+    pub static ref BASELINE_EXPONENT: StoragePower =
+        StoragePower::from_str("340282722551251692435795578557183609728").unwrap();
+    /// 1EiB
+    pub static ref BASELINE_INITIAL_VALUE: StoragePower = StoragePower::from(1) << 60;
+    /// 1EiB
+    pub static ref INIT_BASELINE_POWER: StoragePower =
+    ((BASELINE_INITIAL_VALUE.clone() << (2*PRECISION)) / &*BASELINE_EXPONENT) >> PRECISION;
+    /// 330M for testnet
+    static ref SIMPLE_TOTAL: BigInt = BigInt::from(330_000_000) * FILECOIN_PRECISION;
+    /// 770M for testnet
+    static ref BASELINE_TOTAL: BigInt = BigInt::from(770_000_000) * FILECOIN_PRECISION;
+    /// expLamSubOne = e^lambda - 1
+    /// for Q.128: int(expLamSubOne * 2^128)
+    static ref EXP_LAM_SUB_ONE: BigInt = BigInt::from(37396273494747879394193016954629u128);
+    /// lambda = ln(2) / (6 * epochsInYear)
+    /// for Q.128: int(lambda * 2^128)
+    static ref LAMBDA: BigInt = BigInt::from(37396271439864487274534522888786u128);
+}
+
+/// Compute BaselinePower(t) from BaselinePower(t-1) with an additional multiplication
+/// of the base exponent.
+pub(crate) fn baseline_power_from_prev(prev_power: &StoragePower) -> StoragePower {
+    (prev_power * &*BASELINE_EXPONENT) >> PRECISION
+}
+
+/// Computes RewardTheta which is is precise fractional value of effectiveNetworkTime.
+/// The effectiveNetworkTime is defined by CumsumBaselinePower(theta) == CumsumRealizedPower
+/// As baseline power is defined over integers and the RewardTheta is required to be fractional,
+/// we perform linear interpolation between CumsumBaseline(⌊theta⌋) and CumsumBaseline(⌈theta⌉).
+/// The effectiveNetworkTime argument is ceiling of theta.
+/// The result is a fractional effectiveNetworkTime (theta) in Q.128 format.
+pub(crate) fn compute_r_theta(
+    effective_network_time: ChainEpoch,
+    baseline_power_at_effective_network_time: &BigInt,
+    cumsum_realized: &BigInt,
+    cumsum_baseline: &BigInt,
+) -> BigInt {
+    if effective_network_time != 0 {
+        let reward_theta = BigInt::from(effective_network_time) << PRECISION;
+        let diff = ((cumsum_baseline - cumsum_realized) << PRECISION)
+            / baseline_power_at_effective_network_time;
+
+        reward_theta - diff
+    } else {
+        Default::default()
+    }
+}
+
+/// Computes a reward for all expected leaders when effective network time changes
+/// from prevTheta to currTheta. Inputs are in Q.128 format
+pub(crate) fn compute_reward(
+    epoch: ChainEpoch,
+    prev_theta: BigInt,
+    curr_theta: BigInt,
+) -> TokenAmount {
+    let mut simple_reward = SIMPLE_TOTAL.clone() * &*EXP_LAM_SUB_ONE;
```

Comment:
ah, I actually don't fully understand how this works (seems like it would compile to the same thing because the static would have to be reallocated anyway) But I will make this change here and a line below in case it's more optimized